File: ConsorcioGestFront/ConsorcioGestWeb/src/app/Models/DTO/FiltersDTO.ts
import { Utils } from "src/app/Helpers/Utils";

export class BaseFilter {
  dateFrom?: string;
  dateTo?: string = Utils.parseDate(new Date());
}

//CLAIMS
export class BaseFilterClaimDTO extends BaseFilter {
  causeClaim: number = 0;
  nroReclamo?: string = '';
}

export class FilterClaimDTO extends BaseFilterClaimDTO {
  stateID: number = 0;
}

export class FilterClaimUserDTO extends BaseFilterClaimDTO {
}


//RESERVATIONS
export class BaseFilterReservationsDTO extends BaseFilter {
  commonSpaceID: number = 0;
}

export class FilterReservationDTO {
  dateFrom?: string;
  dateTo?: string;
  document: number = 0;
  commonSpaceID: number = 0;
}

export class FilterReservationUserDTO {
  dateFrom?: string;
  dateTo?: string;
  commonSpaceID: number = 0;
}

export class FilterUserDTO{

  document: number = 0;
  tower:string = '';
  codominium:string = '';
  userStateID: number = 0;
}

export class FilterSurveyDTO extends BaseFilter {
  stateID: number = 0;
  claimNumber: string = '';
}

File: ConsorcioGestBack/BusinessService/DTO/FiltersDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BusinessService.DTO
{
    public class BaseFilter 
    {
        public DateTime? DateFrom { get; set; }
        public DateTime? DateTo { get; set; } = DateTime.Now.Date;
    }

    public class BaseFilterClaimDTO : BaseFilter
    {
        public int CauseClaim { get; set; }
        public string? NroReclamo { get; set; }
    }

    public class FilterClaimDTO : BaseFilterClaimDTO
    {
        public int StateID { get; set; }
    }

    public class FilterClaimUserDTO : BaseFilterClaimDTO
    {

    }

    public class BaseFilterReservationsDTO : BaseFilter
    {
        public int CommonSpaceID { get; set; }
    }

    public class FilterReservationDTO : BaseFilterReservationsDTO 
    {
        public int Document {  get; set; }
    }

    public class FilterReservationUserDTO : BaseFilterReservationsDTO
    {

    }

    public class FilterUserDTO
    {
        public int Document { get; set; }
        public string? Tower { get; set; }
        public string? Condominium { get; set; }
        public int UserStateID { get; set; }
    }

    public class FilterSurveyDTO : BaseFilter 
    {
        public int StateID { get; set; }
        public string? ClaimNumber { get; set; }
    }



}
